packages/components/src/modal/index.tsx

```diff
@@ -184,6 +188,50 @@ function UnforwardedModal(
 		};
 	}, [ bodyOpenClassName ] );
 
+	const frameRef = useRef< HTMLDivElement >();
+	const [ isAnimatingOut, setIsAnimatingOut ] = useState( false );
+	const isReducedMotion = useReducedMotion();
+	const animateClose = useCallback< typeof onRequestClose >(
+		( event ) => {
+			function onAnimationEnd( e: AnimationEvent ) {
```

Comment:
Are you planning on abstracting this to some kind of reusable hook when we need exit animations on other components?

Reply:
Yes, if needed. We could abstract the following:

- the hook takes a boolean prop (ie. "visible", or "open")
- the hook returns a series of props to be applied to the DOM element and potentially a flag to conditionally render the component in the DOM tree
- internally, the hook adds/removes classnames/data attributes based on the "visible" prop changing.
- when the visible prop passes from "true" to "false", the hook will start the exit animation, and only when the animation is finished, will set to `false` the flag to conditionally render the contents in the DOM 

How does that sound ?

Reply:
Sounds good. Also, it might be premature to immediately abstract it to a _reusable_ hook, but I still think there is some value in extracting it out of the main component code, even if it's only used in this component. The logic isn't complicated per se, but it's still a lot of lines for a single and relatively generic purpose.

Reply:
I've moved the code to a separate hook, we can take it from there if it needs more abstraction